src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -26,9 +26,43 @@
 
 import org.spongepowered.api.CatalogType;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the ascending direction, if this rail direction is ascending.
+     *
+     * @return The ascending direction, if possible
+     */
+    Optional<Direction> getAscendingDirection();
+
+    /**
+     * Checks if this rail direction is ascending.
+     *
+     * @return <b>true</b> if is ascending, else <b>false</b>
+     */
+    default boolean isAscending() {
```

Comment:
No reason for the default, can simply do an isPresent().

Reply:
Should I remove default or remove this method?

Reply:
Since ascending direction is optional, remove the default method `isAscending`